src/diffusers/loaders.py

```diff
@@ -294,3 +305,307 @@ def save_function(weights, filename):
         save_function(state_dict, os.path.join(save_directory, weight_name))
 
         logger.info(f"Model weights saved in {os.path.join(save_directory, weight_name)}")
+
+
+class TextEncoderLoRAMixin:
+    r"""
+    This class is used for handling the text encoder used in our pipelines with LoRA. The methods of this class are
+    mostly copy-pasted from [`~UNet2DConditionLoadersMixin`]. We couldn't fully reuse the class because we cannot do
+    things like `self.set_attn_processor()`.
+
+    Args:
+        text_encoder (`nn.Module`):
+            The text encoder module underlying a [`~DiffusionPipeline`].
+    """
+
+    def __init__(self, text_encoder: nn.Module):
```

Comment:
+1, `mixins` should ideally not have `__init__`

Reply:
I don't envision this Mixin to be used with a `DiffusionPipeline` class. 

`_initialize_lora_layers()` initializes the LoRA parameters, and having it inside `load_attn_procs()` is not a good choice IMO since syntactically both of them are doing different things.

Reply:
**Disclaimer: I'm here thinking more about inference of text encoder loras then training**

I think we should have a function called:

```py
load_lora()
```

or:

```py
load_lora_weights(...)
```

That can be called from `StableDiffusionPipeline(...)` 

I don't think we should wrap the text encoder:
```py
text_encoder_lora_wrapper = TextEncoderLoRAMixin(text_encoder)
```

=> this breaks things for **inference**
 `text_encoder_lora_wrapper` cannot be passed to the StableDiffusionPipeline because it doesn't have a forward method, it cannot be saved etc...